packages/flutter/lib/src/widgets/routes.dart

```diff
@@ -971,11 +971,16 @@ abstract class PopupRoute<T> extends ModalRoute<T> {
 /// A [Navigator] observer that notifies [RouteAware]s of changes to the
 /// state of their [Route].
 ///
+/// ## Sample code
+///
 /// To make a [StatefulWidget] aware of its current [Route] state, implement
 /// [RouteAware] in its [State] and subscribe it to the [RouteObserver]:
 ///
 /// ```dart
 /// class RouteAwareWidgetState extends State<RouteAwareWidget> with RouteAware {
+///
+///   final RouteObserver routeObserver = new RouteObserver();
```

Comment:
We should make that clear in the sample code then. You should assume that the sample code is what will be copied-and-pasted. It should be the most pedantically correct thing, because it represents the best that we can hope to see in deployed code.